components/tel/Core/Src/canload.c

```diff
@@ -0,0 +1,72 @@
+#include "canbusload.h"
+
+uint32_t can_total_bits = 0;
+static uint64_t counter_window = 0;
+static uint64_t current_bus_load = 0;
+uint32_t currentIdx = 0;
+uint32_t circularBuffer[CANLOAD_WINDOW_SIZE] = {0};
+
+uint32_t CANLOAD_calculate_message_bits(CAN_msg_t *msg)
```

Comment:
This is another public function to be included in the TEL interrupt when new messages are received and it keeps a running total of all bits sent over the specified windwo